### DISCUSSION TITLE
Documentation Failed to fetch /openapi.json

### DISCUSSION THREAD

**Discussion by NastasiaSaby** [fastapi/fastapi#7395] [Questions] [answered]
When I click https://host/suburl/documentation

I have a "Failed to load definition". 

In the console, I have

```
Error downloading https://host/openapi.json Failed to fetch
```

My code looks like that:

```
from fastapi import FastAPI

app = FastAPI(openapi_url="/openapi.json", docs_url="/documentation")

@app.get("/status")
def return_simple():
    return "ok"
I can't find an anwser for this problem. Do you have any idea?

```

**Comment by NastasiaSaby** [comment, marked as answer]
I found a solution. I launched fastapi with a "root-path" like this:

```
uvicorn fast_api.server:app --host 0.0.0.0 --port 8000 --root-path="/suburl"
```

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:

**Comment by SirBenedick** [comment]
> I found a solution. I launched fastapi with a "root-path" like this:
> 
> ```
> uvicorn fast_api.server:app --host 0.0.0.0 --port 8000 --root-path="/suburl"
> ```

fyi: it is also possible to pass the root_path as a parameter while creating a FastAPI instance (as described [here](https://fastapi.tiangolo.com/advanced/behind-a-proxy/?h=root+path#setting-the-root_path-in-the-fastapi-app)):

```
from fastapi import FastAPI, Request

app = FastAPI(root_path="/api/v1")


@app.get("/app")
def read_main(request: Request):
    return {"message": "Hello World", "root_path": request.scope.get("root_path")}
```

**Reply by Haleshot** [reply]
> I found a solution. I launched fastapi with a "root-path" like this:
> ```
> uvicorn fast_api.server:app --host 0.0.0.0 --port 8000 --root-path="/suburl"
> ```

Just wanted to add on to the path mentioned above:
If your main app files (for running the fast api server like `server` mentioned above) do not lie in the root directory, then modify the command above as follows:

- Say the app running Fast API is called main.py and lies inside the src folder

```sh 
uvicorn src.main:app --host 0.0.0.0 --port 8000 --root-path="/suburl"
```

This might be a known case but just putting it here in case someone finds it useful.

**Comment by joshb019919** [comment]
Thank you, Sir Benedick!  My career as a student has taken me away from
Fast API for the moment, but should I return (and I'm intrigued by the
whole process) I'll refer back to this.  Have a great day.
Josh
